### PR TITLE
alias: Add option to use relative alias offset

### PR DESCRIPTION
**Pull request by agkaminski**
DONE: RTOS-713

## Description

**OPTIONAL** - to be determined whether we want this or not

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (ia32-generic-qemu).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: cmds/alias.c [hunk 1]
```diff
@@ -30,38 +35,79 @@ static int cmd_alias(int argc, char *argv[])
...
+		if (c < 0) {
+			break;
+		}
+
+		switch (c) {
+			case 'r':
+				relative = 1;
+				break;
+
+			case 'b':
```
Comment (gorbanchovie) [inline cmds/alias.c:58]: `argc` could be checked. I think it should be, as it's not obvious that `alias -b 0x10000000 test 0x200 0x200` (which returns no error) doesn't create alias

File: cmds/alias.c [hunk 2]
```diff
@@ -17,11 +17,16 @@
...
+#include <lib/log.h>
+
+
+static addr_t aliasBase = 0;
 
 
 static void cmd_aliasInfo(void)
 {
-	lib_printf("sets alias to file, usage: alias [<name> <offset> <size>]");
+	lib_printf("sets alias to file, usage: alias [<-r> <-b BASE>] [<name> <offset> <size>]");
```
Comment (gorbanchovie) [inline cmds/alias.c:29]: I don't know what are usage writing guidelines, but I think it should look like this: `[-b <base> | [-r] <name> <offset> <size>]`

File: cmds/alias.c [hunk 3]
```diff
@@ -30,38 +35,79 @@ static int cmd_alias(int argc, char *argv[])
...
+		log_error("\n%s: Wrong arguments", argv[0]);
+		return CMD_EXIT_FAILURE;
+	}
 
-		if (*end) {
-			log_error("\n%s: Wrong arguments", argv[0]);
-			return -EINVAL;
+	if (relative != 0) {
+		if (addr + aliasBase < addr) {
+			log_error("\n%s: Relvative offset overflow", argv[0]);
```
Comment (gorbanchovie) [inline cmds/alias.c:97]: typo

File: cmds/alias.c (hunk 1 above)
Comment (agkaminski) [reply cmds/alias.c:58]: Done

File: cmds/alias.c (hunk 2 above)
Comment (agkaminski) [reply cmds/alias.c]: Way better, thanks, I always have a problem with these

File: cmds/alias.c (hunk 3 above)
Comment (agkaminski) [reply cmds/alias.c]: Fixed

**Review by gorbanchovie — APPROVED** [review]
lgtm

**Review by gerard5 — COMMENTED** [review]
I'd add small log improvements, see inline comments.

File: cmds/alias.c [hunk 4]
```diff
@@ -30,38 +35,85 @@ static int cmd_alias(int argc, char *argv[])
 	char *end;
 	size_t sz = 0;
 	addr_t addr = 0;
-	unsigned int i;
+	int c, relative = 0;
+	addr_t newBase;
 
 	if (argc == 1) {
+		lib_printf("\nRelative alias base: 0x%p", aliasBase);
```
[the remark this answers is not in the document]
Comment (agkaminski) [reply cmds/alias.c]: Done

File: cmds/alias.c [hunk 5]
```diff
@@ -30,38 +35,85 @@ static int cmd_alias(int argc, char *argv[])
...
+
+				newBase = lib_strtoul(optarg, &end, 0);
+				if (*end != '\0') {
+					log_error("\n%s: Invalid base.\n", argv[0]);
+					cmd_aliasInfo();
+					return CMD_EXIT_FAILURE;
+				}
+
+				aliasBase = newBase;
+
```
[the remark this answers is not in the document]
Comment (agkaminski) [reply cmds/alias.c:76]: Done

**Review by gorbanchovie — APPROVED** [review]
lgtm

**Comment by kemonats** [comment]
This PR has been merged but I have a minor comment, in lines `113` and `117` instead of `argv[1]` it should be `argv[optind]`.

**Pull request merged**